### PR TITLE
General changes and updates

### PR DESCRIPTION
**Pull request by JancoDVTDev**
## General changes and updates

- Gamma Slider on photo editor fixed with range limiter.
- Filter chaining implemented.
- Undo button functionality.
- Trash Button changed to Red.
- Cropping feature updated.

### CODE REVIEW

File: camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift [hunk 1]
```diff
@@ -176,11 +176,11 @@ public class PhotoEditorViewModel {
         let scaleFactorY = CGFloat(cgCropImage!.height)/scaledHeight
 
         if originalCGCropImageWidth > originalCGCropImageHeight {
-            scaledOriginY = (cropView.frame.origin.y - (imageViewHeight - scaledHeight)) * scaleFactorY
+            scaledOriginY = (CGFloat(imageViewHeight - scaledHeight!)) * scaleFactorY
```
Comment (NathanDVT) [inline camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift:179]: why force '!' variable value, if you use it in 176 without forcing a value
Naturally forces should be prevented as they could crash your demo and Codacy should give you a warning

File: camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift [hunk 2]
```diff
@@ -236,7 +236,23 @@ public class PhotoEditorViewModel {
...
+            rectWidth! /= 3
+            rectWidth! *= 2
+            rectHeight! /= 3
+            if isLandscape {
+                rectOriginX = CGFloat(widthMiddle) - CGFloat(Int(rectWidth!/2))
+                rectOriginY = CGFloat(imageView.frame.size.height) - scaledHeight
+            } else {
+                rectOriginX = imageView.frame.origin.x + (imageView.frame.size.width - scaledWidth)/2
+                rectOriginY = heightMiddle - scaledHeight/5
+            }
```
Comment (NathanDVT) [inline camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift]: I would suggest maybe use a guard statement so you won't have as many forced non-null instances

**Review by NathanDVT — APPROVED** [review]
LGTM

File: camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift (hunk 1 above)
Comment (JancoDVTDev) [reply camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift:179]: Codacy does not warn, the variable does not contain the right value when unforced

File: camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift (hunk 2 above)
Comment (JancoDVTDev) [reply camPod/Classes/EditPhoto/VIewModel/PhotoEditorViewModel.swift]: Cool, will change that, maybe it fixes the problem I have

**Review by XanderSchoeman — APPROVED** [review]

**Pull request merged**